### DISCUSSION TITLE
How to bundle @ruffle-rs/ruffle npm package with webpack?

### DISCUSSION THREAD

**Discussion by orazioedoardo** [ruffle-rs/ruffle#18569] [Q&A]
I'm trying to use the NPM package to centralize javascript dependencies to NPM and benefit from dependabot. Ruffle works fine when including the release in a script tag, however I can't figure out how I'm supposed to use the `@ruffle-rs/ruffle` package. Suppose a `player.js` file:

```
import RufflePlayer from "@ruffle-rs/ruffle";

const ruffle = window.RufflePlayer.newest();
```

Then create the bundle. Works but just like the release, it tries to load `core.ruffle.somehash.js` and then I suppose the WebAssembly files. Should I just manually copy those files from the node_modules folder? https://ruffle.rs/demo/ doesn't seem to do this.

**Comment by ColonelThirtyTwo** [comment]
As far as I can tell, `@ruffle-rs/ruffle` is the `ruffle-standalone` package that writes its API to `window.RufflePlayer` and applies the polyfill but does not actually export anything from its JS module.

The package mentions: "If you'd prefer to use Ruffle through npm and a bundler, please [refer to ruffle core](https://github.com/ruffle-rs/ruffle/tree/master/web/packages/core)."... but that package isn't published anywhere, so it's pretty much impossible to use. The `ruffle-demo` uses it, but can do so only because it's sharing a workspace with `ruffle-core`.

Ultimately there really ought to be a `@ruffle-rs/ruffle-core` published alongside the `@ruffle-rs/ruffle` package.
